v2/metadata/client.go

```diff
@@ -124,13 +125,16 @@ type Execution struct {
 
 // GetPipeline returns the current pipeline represented by the specified
 // pipeline name and run ID.
-func (c *Client) GetPipeline(ctx context.Context, pipelineName string, pipelineRunID string) (*Pipeline, error) {
-	pipelineContext, err := getOrInsertContext(ctx, c.svc, pipelineName, pipelineContextType)
+func (c *Client) GetPipeline(ctx context.Context, pipelineName, pipelineRunID, namespace, runResource string) (*Pipeline, error) {
+	pipelineContext, err := getOrInsertContext(ctx, c.svc, pipelineName, pipelineContextType, nil)
 	if err != nil {
 		return nil, err
 	}
-
-	pipelineRunContext, err := getOrInsertContext(ctx, c.svc, pipelineRunID, pipelineRunContextType)
+	runMetadata := map[string]*pb.Value{
+		"namespace":     stringValue(namespace),
+		"resource_name": stringValue(runResource),
+	}
+	pipelineRunContext, err := getOrInsertContext(ctx, c.svc, pipelineRunID, pipelineRunContextType, runMetadata)
```

Comment:
Since we change the value of "pipelineRunID" here, from workflow id to the uuid of the pipelineRun, will user still be able to get info of the previous tasks(which stores the workflow id in MLMD) via UI?

Reply:
Good catch! No, they won't. This is under the assumption that v2 compatible doesn't have many users, we kept saying it's experimental and we can make breaking changes.